waspc/src/Analyzer.hs

```diff
@@ -0,0 +1,25 @@
+module Analyzer
+  ( analyze,
+  )
+where
+
+import Analyzer.Decl (Decl)
+import qualified Analyzer.Evaluator as E
+import qualified Analyzer.Parser as P
+import Analyzer.StdLib (stdLib)
+import qualified Analyzer.TypeChecker as T
+import Control.Arrow (left)
+import Control.Monad ((>=>))
+
+data AnalyzeError
+  = ParseError P.ParseError
+  | TypeError T.TypeError
+  | EvaluationError E.EvaluationError
+
+-- | Takes a Wasp source file and produces a list of declarations or a
+--   description of an error in the source file.
+analyze :: String -> Either AnalyzeError [Decl]
```

Comment:
Yes, I plan to have add error recovery in the future. I'm estimating that it will be a fair amount of work to get error recovery working, and I'm prioritizing having a basic working version of the entire analyzer first.